align/pnr/main.py

```diff
@@ -121,48 +123,53 @@ def remove_pg_pins(verilog_d:dict, subckt:str, pg_connections:dict):
+    modules = [module for module in verilog_d['modules'] if module['name'] == subckt]
+    assert len(modules) == 1, f"duplicate modules are found {modules}"
     module = modules[0]
-    #Remove port from subckt level
+    # Remove port from subckt level
     module['parameters'] = [p for p in module['parameters'] if p not in pg_connections.keys()]
     for inst in module['instances']:
         if inst['abstract_template_name'] in modules_dict:
-            #Inst pins connected to pg_pins
-            hier_pg_connections = {conn["formal"]:pg_connections[conn["actual"]] for conn in inst['fa_map'] if conn["actual"] in pg_connections}
-            if len(hier_pg_connections)>0:
+            # Inst pins connected to pg_pins
+            hier_pg_connections = {conn["formal"]: pg_connections[conn["actual"]] for conn in inst['fa_map'] if conn["actual"] in pg_connections}
+            if len(hier_pg_connections) > 0:
                 logger.debug(f"Removing instance {inst['instance_name']} pins connected to {hier_pg_connections}")
                 inst['fa_map'] = [conn for conn in inst['fa_map'] if conn["formal"] not in hier_pg_connections]
-                #Creates a copy of module with reduced pins
+                # Creates a copy of module with reduced pins
                 new_name = modify_pg_conn_subckt(verilog_d, inst['abstract_template_name'], hier_pg_connections)
                 inst['abstract_template_name'] = new_name
                 remove_pg_pins(verilog_d, new_name, hier_pg_connections)
         else:
-            inst['fa_map'] = [{'formal':conn['formal'],
-                        'actual': pg_connections.get(conn["actual"], conn["actual"])} for conn in inst['fa_map']]
+            inst['fa_map'] = [{'formal': conn['formal'],
+                               'actual': pg_connections.get(conn["actual"], conn["actual"])} for conn in inst['fa_map']]
             logging.debug(f"leaf level node {inst['instance_name']} {inst['abstract_template_name']}")
 
+
 def clean_if_extra(verilog_d, subckt):
     #Remove modules which are not instantiated
     all_inst = set([inst['abstract_template_name'] for module in verilog_d['modules'] for inst in module["instances"]]+[subckt])
     logger.debug(f"All used modules: {all_inst}")
     verilog_d['modules'] = [m for m in verilog_d['modules'] if m['name'] in all_inst]
 
+
 def modify_pg_conn_subckt(verilog_d, subckt, pp):
     """
-    creates a new subcircuit by removing power pins from a subcircuit defination
+    creates a new subcircuit by removing power pins from a subcircuit definition
     and change internal connections within the subcircuit
     nm: new module
     """
-    #TODO: remove redundant const
-    nm = deepcopy([module for module in verilog_d['modules'] if module['name']==subckt][0])
+    # TODO: remove redundant const
+
+    nm = copy.copy([module for module in verilog_d['modules'] if module['name'] == subckt][0])
```

Comment:
@stevenmburns , is shallow copy okay to do here as `VerilogJsonModule` has mutable attributes?